### ISSUE TITLE
new theme needs progress UI

### ISSUE THREAD

**Issue by peller** [maqetta/maqetta#944]
@billreed63

Hit OK on the new theme dialog and there is a delay while the new theme is created.  Meanwhile, the dialog remains up and you can click on the OK button again (and get an error message).  The dialog should probably be dismissed and some sort of loading... UI put up. Alternatively, we could disable the dialog buttons.

**Comment by billreed63** [comment]
I have spent a good bit of time looking into this, the issue that we have is the logic that clones a theme has several steps. I have found that the browser does not update the UI until the browser thread is released. In the case of theme cloning that means that the user does not see any of the progress messages.  
One solution to this issue is to move the cloning theme logic into a webWorker, although this may not be possible with all the dojo code that is involved in the Model code [look at dojo ticket 11750 Support for Dojo in WebWorkers](http://bugs.dojotoolkit.org/ticket/11750) 
Another option might be to move the clone theme into a setTimeout to release the browser thread, but each step of the clone theme would need to be in a setTimeout.

We could just put the spinning disc up at the start of the clone and remove it at the end.. not very helpful, but better than nothing I guess...

**Comment by peller** [comment]
Changing long-running blocking I/O commands to async should solve the problem.  The async callbacks, like a setTimeout, will allow the stack to unwind, and I/O can run cooperatively with the progress UI.

**Comment by billreed63** [comment]
The I/O is async, the problem is not I/O blocking it is that the javaScript is  updating the almost every CSSRule in the CSSModle this is CPU intensive and is blocking the browser from updating the UI with progress updates.  I have tried placing progress updates at almost every point in the clone process including in the deferred.then for each file I/O. I have found that the UI is not updated until the clone is complete, or if I put break points and single step through the code. I would assume because performance enhancements of browser
